src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -508,11 +513,52 @@ public void handleFlowerPots(@NotNull final BlockPos pos)
     }
 
     @Override
+    public void handleBuildingOverChest(@NotNull final BlockPos pos, final TileEntityChest chest)
+    {
+        if(job.getWorkOrder() != null && getOwnBuilding().getColony().getBuilding(job.getWorkOrder().getBuildingLocation()) instanceof BuildingWareHouse)
+        {
+            final List<ItemStack> inventory = new ArrayList<>();
+            final int size = chest.getSingleChestHandler().getSlots();
+            for(int slot = 0; slot < size; slot++)
+            {
+                final ItemStack stack = chest.getSingleChestHandler().getStackInSlot(slot);
+                if(!ItemStackUtils.isEmpty(stack))
+                {
+                    inventory.add(stack.copy());
+                }
+                chest.getSingleChestHandler().extractItem(slot, Integer.MAX_VALUE, false);
+            }
+
+            world.setBlockState(pos, ModBlocks.blockRack.getDefaultState(), 0x03);
+            final TileEntity entity = world.getTileEntity(pos);
+            if(entity instanceof TileEntityRack)
+            {
+                for(final ItemStack stack: inventory)
+                {
+                    if(!ItemStackUtils.isEmpty(stack))
+                    {
+                        InventoryUtils.addItemStackToItemHandler(((TileEntityRack) entity).getInventory(), stack);
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
     public void connectBlockToBuildingIfNecessary(@NotNull Block block, @NotNull final BlockPos pos)
     {
         final BlockPos buildingLocation = job.getWorkOrder().getBuildingLocation();
         final AbstractBuilding building = this.getOwnBuilding().getColony().getBuilding(buildingLocation);
 
+        if(building instanceof BuildingWareHouse && block instanceof BlockChest)
```

Comment:
This is more for legacy.
We need to replace all the building schematics, but since this is a huge mess to rescan all of them.
We'll leave this here inside until we updated all of them.
Then we can remove this here again.